src/main/java/openblocks/common/PerlinNoiseGenerator.java

```diff
@@ -0,0 +1,635 @@
+package openblocks.common;
+
+import java.util.Random;
+
+
+/*****************************************************************************
+ *						  J3D.org Copyright (c) 2000
+ *								 Java Source
+ *
+ * This source is licensed under the GNU LGPL v2.1
+ * Please read http://www.gnu.org/copyleft/lgpl.html for more information
+ *
+ * This software comes with the standard NO WARRANTY disclaimer for any
+ * purpose. Use it at your own risk. If there's a problem you get to fix it.
+ *
+ ****************************************************************************/
+
+
+/**
+ * Computes Perlin Noise for three dimensions.
+ * <p>
+ *
+ * The result is a continuous function that interpolates a smooth path
+ * along a series random points. The function is consitent, so given
+ * the same parameters, it will always return the same value. The smoothing
+ * function is based on the Improving Noise paper presented at Siggraph 2002.
+ * <p>
+ * Computing noise for one and two dimensions can make use of the 3D problem
+ * space by just setting the un-needed dimensions to a fixed value.
+ *
+ * @author Justin Couch
+ * @version $Revision: 1.4 $
+ */
+public class PerlinNoiseGenerator
```

Comment:
This was bugging me, too. I'll have to do some testing with that since I did some serious comparision of different noise algorithms to get the right "feeling" when flying. I did this using 2-D visualisation, so I'll have to see how to do the same with internal minecraft functions.